flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaConsumerTestBase.java

```diff
@@ -1217,6 +1219,99 @@ public void flatMap(Tuple3<Integer, Integer, String> value, Collector<Integer> o
 		}
 	}
 
+	/**
+	 * Test producing and consuming into multiple topics.
+	 * @throws Exception
+	 */
+	public void runProduceConsumeMultipleTopicsWithKafkaSerializer() throws Exception {
```

Comment:
Can we avoid duplicating the test code by doing the following?
1. have a private method taking schema as an argument.
2. let `runProduceConsumeMultipleTopicsWithKafkaSerializer` and `runProduceConsumeMultipleTopicsWithLegacySerializer` call that method and pass in different schema?

Reply:
How do you mean? The two schemas don't share a common base class (except `Object`).

Reply:
I was thinking of using something like `instanceOf`. Given it is private method, we don't need to worry much about the wrong argument type. We may need to add some comments to explain that. Admittedly, this is not ideal, but personally I feel it is slightly better than having a lot of duplicate code.